Add UI for CreateRaffle and AddPrize

gabrielhicks opened the pull request:
I modified the Admin Screen to include additional "+" signs and information to create new raffles, and once created, add new prizes to those raffles.



TODO:
- Think about how to handle the Raffle Whitelist in a more dynamic way
- Add Close Entrants API to reclaim rent on Entrants accounts

gabrielhicks committed:
Fix small for loop error

gabrielhicks committed:
Merge branch 'draffle-io:master' into master

gabrielhicks committed:
Implement Create Raffle, Add Prize, and re-style Admin Home Screen to play nice with changes

gabrielhicks committed:
Add comments, remove logs

gabrielhicks commented:
Addresses issue #15

jdomingospt commented:
Hello @gabrielhicks I've been testing your code, althoug i'm not being able to create a raffle, I keep getting this response:

`0: "Program 11111111111111111111111111111111 invoke [1]"
1: "Program 11111111111111111111111111111111 success"
2: "Program dRafA7ymQiLKjR5dmmdZC9RPX4EQUjqYFB3mWokRuDs invoke [1]"
3: "Program 11111111111111111111111111111111 invoke [2]"
4: "Program 11111111111111111111111111111111 success"
5: "Program 11111111111111111111111111111111 invoke [2]"
6: "Program 11111111111111111111111111111111 success"
7: "Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA invoke [2]"
8: "Program log: Instruction: InitializeAccount"
9: "Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA consumed 3602 of 1367665 compute units"
10: "Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA success"
11: "Program log: libstd rust_begin_panic"
12: "Program log: panicked at 'from_bytes_mut>SizeMismatch', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/bytemuck-1.7.2/src/lib.rs:115:3"
13: "Program dRafA7ymQiLKjR5dmmdZC9RPX4EQUjqYFB3mWokRuDs consumed 51581 of 1400000 compute units"
14: "Program failed to complete: BPF program panicked"
15: "Program dRafA7ymQiLKjR5dmmdZC9RPX4EQUjqYFB3mWokRuDs failed: Program failed to complete"`

I'm using a wallet that has authority over that token, I tried to tweak your code a bit but I ran out of "solutions".
Any ideas ?

gabrielhicks committed:
Added DateTimePicker, Close Entrants, Collect Proceeds, and a table for raffle tickets/wallets

gabrielhicks commented:
> Hello @gabrielhicks I've been testing your code, althoug i'm not being able to create a raffle, I keep getting this response:
> 
> `0: "Program 11111111111111111111111111111111 invoke [1]" 1: "Program 11111111111111111111111111111111 success" 2: "Program dRafA7ymQiLKjR5dmmdZC9RPX4EQUjqYFB3mWokRuDs invoke [1]" 3: "Program 11111111111111111111111111111111 invoke [2]" 4: "Program 11111111111111111111111111111111 success" 5: "Program 11111111111111111111111111111111 invoke [2]" 6: "Program 11111111111111111111111111111111 success" 7: "Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA invoke [2]" 8: "Program log: Instruction: InitializeAccount" 9: "Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA consumed 3602 of 1367665 compute units" 10: "Program TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA success" 11: "Program log: libstd rust_begin_panic" 12: "Program log: panicked at 'from_bytes_mut>SizeMismatch', /root/.cargo/registry/src/github.com-1ecc6299db9ec823/bytemuck-1.7.2/src/lib.rs:115:3" 13: "Program dRafA7ymQiLKjR5dmmdZC9RPX4EQUjqYFB3mWokRuDs consumed 51581 of 1400000 compute units" 14: "Program failed to complete: BPF program panicked" 15: "Program dRafA7ymQiLKjR5dmmdZC9RPX4EQUjqYFB3mWokRuDs failed: Program failed to complete"`
> 
> I'm using a wallet that has authority over that token, I tried to tweak your code a bit but I ran out of "solutions". Any ideas ?

Hey! I am not exactly sure where the size mismatch is happening, however I have added some changes to make the UI more intuitive.
- [x] Added 3 hour, 6 hour, 1 day, 3 days, 1 week button options for creating raffle end timestamps
- [x] Added DateTimePicker component to pull local time and convert to Epoch Time
- [x] Added a multiplier for token cost (it assumes 9 decimal tokens)
- [x] Added a sort to display raffles by most recently ended in the Admin Panel
- [x] Added ability to collect raffle proceeds in the Admin Raffle View
- [x] Added ability to close entrants account (reclaim SOL from raffle) in the Admin Raffle View
- [x] Added button to end raffle early in the Admin Raffle View
- [x] Added a small function to the program as well as the CLI to enable the ability to close raffles early

gabrielhicks committed:
Update Card mis-click error

gabrielhicks committed:
Update keys

gabrielhicks committed:
Update key prop for navbuttons

gabrielhicks committed:
Remoev personal wallet

gabrielhicks committed:
Merge updates to CLI, program, README

gabrielhicks commented:
Ill be adding quite a bit of changes and updates "soon"
